tests/test-buttercup.el

```diff
@@ -881,7 +894,52 @@
       (it "throws an error when called"
         (expect (test-function 1 2)
                 :to-throw
-                'error "Stubbed error")))))
+                'error "Stubbed error")))
+
+    (describe "error-recording functionality"
+      (before-each
+        (spy-on 'test-function-throws-on-negative :and-call-through))
+
+      (it "records the function as called even if it throws an error"
+        (expect (test-function-throws-on-negative -5) :to-throw)
+        (expect (buttercup--apply-matcher
```

Comment:
Why are you not using `(expect 'test-function-throws-on-negative :to-have-been-called)` here?
Same for the cases below.

Reply:
If you look above, many of the existing test cases do likewise. I don't recall the reason at the moment, but I vaguely remember that there *is* a reason. In any case, I'm just following the pattern.